### PR TITLE
JS password validation doesn't work as intended with uppercase and lowercase minimum requirements

### LINKED ISSUE #34151: JS password validation doesn't work as intended with uppercase and lowercase minimum requirements
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

login/ui

### Describe the bug

When having a minimum number of uppercase and lowercase characters required for new passwords, they're not correctly validated during registration. 

The code doesn't check if there are enough characters of the required type, but if the amount of characters of all other types is bigger than the set minimum.

Uppercase:
`value.split("").filter((char) => char !== char.toUpperCase()).length > policy.value`

Lowercase:
`value.split("").filter((char) => char !== char.toLowerCase()).length > policy.value`

### Version

26.0.0

### Regression

- [ ] The issue is a regression

### Expected behavior

No error messages are shown if the password contains the minimum amount of uppercase and lowercase characters.

### Actual behavior

Error messages are shown despite the password containing the minimum amount of uppercase and lowercase characters.

### How to Reproduce?

- Use any realm with registration enabled
- Go to Authentication -> Policies -> Password policy and enable uppercase chars & lowercase chars and click on Save
- Go to the registration page of the realm
- Type `TEst` in the password field and leave it to let the JavaScript code validate it with the policies
  - It results in error messages
- Type `1234` in the password field and leave it to let the JavaScript code validate it with the policies
  - It results in no errors at all

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SebastEnn**
Closes #34151

Signed-off-by: SebastEnn <103125747+SebastEnn@users.noreply.github.com>

### CODE REVIEW

File: themes/src/main/resources/theme/keycloak.v2/login/resources/js/password-policy.js [hunk 1]
```diff
@@ -11,15 +11,15 @@ const policies = {
   },
   upperCase: (policy, value) => {
     if (
-      value.split("").filter((char) => char !== char.toUpperCase()).length >
+      value.split("").filter((char) => char === char.toUpperCase() && char !== char.toLowerCase()).length <
```
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/login/resources/js/password-policy.js:14]: If `char` is the same as `char.toUpperCase()` then why would we still need to check the lowercase? Same for the other condition, but inversed.

Comment (jonkoops) [reply themes/src/main/resources/theme/keycloak.v2/login/resources/js/password-policy.js:14]: I see, perhaps then a regular expression would be better suited for this? For example:

```js
value.match(/[A-Z]/g)?.length ?? 0; // This should give a count of all uppercase alpha characters.
```

Comment (SebastEnn) [reply themes/src/main/resources/theme/keycloak.v2/login/resources/js/password-policy.js:14]: But afaik characters like ä, ö and ü are not included in `[a-z]`.

Comment (jonkoops) [reply themes/src/main/resources/theme/keycloak.v2/login/resources/js/password-policy.js:14]: Fair enough, let's consider this resolved then.

**Review by jonkoops — APPROVED** [review]
LGTM

**Comment by jonkoops** [comment]
Backport for 26.x up under #34240.

**Pull request merged**